ckan/lib/search/__init__.py

```diff
@@ -1,318 +1,23 @@
-import logging
-import sys
-import cgitb
-import warnings
-import xml.dom.minidom
-import urllib2
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from ckan import plugins
 
-from pylons import config
-from paste.deploy.converters import asbool
 
-import ckan.model as model
-import ckan.plugins as p
-import ckan.logic as logic
+def index(package):
+    for plugin in plugins.PluginImplementations(plugins.IIndexer):
+        return plugin.index(package)
```

Comment:
If we only expect that a single `IIndexer` implementation is active at a given time then I'd prefer
```python
indexer = list(plugins.PluginImplementations(plugins.IIndexer))[0]
return indexer.index(package)
```
This will at least throw an error if no indexer is registered.

If multiple indexers are to be supported then I'd suggest
```python
for plugin in plugins.PluginImplementations(plugins.IIndexer):
    plugin.index(package)
```
altough it's not clear what the correct return value would be in that case.